src/Tribe/Assets.php

```diff
@@ -0,0 +1,338 @@
+	 *
+	 * @var array
+	 */
+	private $assets = array();
+
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self;
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Register the Methods in the correct places
+	 */
+	private function __construct() {
+		// Hook the actual registering of
+		add_action( 'init', array( $this, 'action_maybe_register' ), 1 );
+	}
+
+	/**
+	 * Register the Assets on the correct hooks
+	 *
+	 * @return void
+	 */
+	public function action_maybe_register() {
+		foreach ( $this->assets as $asset ) {
+			if ( 'js' === $asset->type ) {
+				wp_register_script( $asset->slug, $asset->url, $asset->deps, $asset->version, $asset->in_footer );
+			} else {
+				wp_register_style( $asset->slug, $asset->url, $asset->deps, $asset->version, $asset->media );
+			}
+
+			// If we don't have an action we don't even register the action to enqueue
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			// Now add an action to enqueue the registred assets
+			add_action( $asset->action, array( $this, 'enqueue' ), $asset->priority );
+		}
+	}
+
+	public function enqueue() {
+		foreach ( $this->assets as $asset ) {
+			// Skip if we are not on the correct filter
+			if ( current_filter() !== $asset->action ) {
+				continue;
+			}
+
+			// If here is no action we are just registering this asset
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			/**
+			 * Allows developers to hook-in and prevent an asset from been loaded
+			 *
+			 * @param bool   $enqueue If we should enqueue or not a given asset
+			 * @param object $asset   Which asset we are dealing with
+			 */
+			$enqueue = apply_filters( 'tribe_asset_enqueue', true, $asset );
+
+			/**
+			 * Allows developers to hook-in and prevent an asset from been loaded
+			 *
+			 * Note: When you pass callables on the `$asset->filter` argument this will be hooked here
+			 *
+			 * @param bool   $enqueue If we should enqueue or not a given asset
+			 * @param object $asset   Which asset we are dealing with
+			 */
+			$enqueue = apply_filters( 'tribe_asset_enqueue_' . $asset->slug, $enqueue, $asset );
+
+			if ( ! $enqueue ) {
+				continue;
+			}
+
+			if ( 'js' === $asset->type ) {
+				wp_enqueue_script( $asset->slug );
+
+				// Only localize on JS and if we have data
+				if ( ! empty( $asset->localize ) ) {
+					wp_localize_script( $asset->slug, $asset->localize->name, $asset->localize->data );
+				}
+			} else {
+				wp_enqueue_style( $asset->slug );
+			}
+		}
+	}
+
+	/**
+	 * Returns the path to a minified version of a js or css file, if it exists.
+	 * If the file does not exist, returns false.
+	 *
+	 * @param string $url   The path or URL to the un-minified file.
+	 *
+	 * @return string|false The path/url to minified version or false, if file not found.
+	 */
+	public static function maybe_get_min_file( $url ) {
```

Comment:
Looking at the way this method is used (from within the `register()` method for example), I'm wondering if it would make sense to make this more tolerant of missing minified assets.

For instance, right now if `abc.js` exists but `abc.min.js` does not, it will return false: looking back to that `register()` method that would mean we'd be setting `$asset->url` to false, when actually there might be a valid URL we could be using.